From 333c01659781cb5d956476f5b4cd3366a6f2d81d Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Thu, 20 Jul 2023 09:13:43 -0400
Subject: [PATCH] Add an always-on `inode64` feature

As I (and others) will be backporting the fix in
https://github.com/ostreedev/ostree/pull/2874/commits/de6fddc6adee09a93901243dc7074090828a1912
pretty far, I want a way for sysadmins and OS builders to
be able to reliably see when their version of ostree has this fix
(Because comparing version numbers isn't portable).
---
 configure.ac              | 5 ++++-
 tests/basic-test.sh       | 7 ++++++-
 tests/inst/src/repobin.rs | 5 ++++-
 3 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0e45aa87..9de69551 100644
--- a/configure.ac
+++ b/configure.ac
@@ -82,7 +82,10 @@ AM_COND_IF([BUILDOPT_TSAN],
 LT_PREREQ([2.2.4])
 LT_INIT([disable-static])
 
-OSTREE_FEATURES=""
+dnl We have an always-on feature now to signify the fix for 
+dnl https://github.com/ostreedev/ostree/pull/2874/commits/de6fddc6adee09a93901243dc7074090828a1912
+dnl "commit: fix ostree deployment on 64-bit inode fs"
+OSTREE_FEATURES="inode64"
 AC_SUBST([OSTREE_FEATURES])
 
 GLIB_TESTS
diff --git a/tests/basic-test.sh b/tests/basic-test.sh
index e2a7d70c..0a50ae2f 100644
--- a/tests/basic-test.sh
+++ b/tests/basic-test.sh
@@ -19,7 +19,7 @@
 
 set -euo pipefail
 
-echo "1..$((90 + ${extra_basic_tests:-0}))"
+echo "1..$((91 + ${extra_basic_tests:-0}))"
 
 CHECKOUT_U_ARG=""
 CHECKOUT_H_ARGS="-H"
@@ -41,6 +41,11 @@ fi
 # This should be dynamic now
 assert_not_has_dir repo/uncompressed-objects-cache
 
+if ! $OSTREE --version | grep -q -e '- inode64'; then
+    fatal missing inode64 feature
+fi
+echo "ok inode64 feature"
+
 validate_checkout_basic() {
     (cd $1;
      assert_has_file firstfile
